src/main/kotlin/com/dprint/services/editorservice/EditorProcess.kt

```diff
@@ -0,0 +1,177 @@
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.project.Project
+import java.io.File
+import java.nio.ByteBuffer
+
+private const val BUFFER_SIZE = 1024
+private const val ZERO = 0
+private const val U32_BYTE_SIZE = 4
+
+private val LOGGER = logger<EditorProcess>()
+
+// Dprint uses unsigned bytes of 4x255 for the success message and that translates
+// to 4x-1 in the jvm's signed bytes.
+private val SUCCESS_MESSAGE = byteArrayOf(-1, -1, -1, -1)
+
+class EditorProcess(private val project: Project) {
+    private var process: Process? = null
+
+    fun initialize() {
+        val notificationService = project.service<NotificationService>()
+        val executablePath = FileUtils.getValidExecutablePath(this.project)
+        val configPath = FileUtils.getValidConfigPath(project)
+
+        if (this.process != null) {
+            destroy()
+        }
+
+        when {
+            configPath.isNullOrBlank() -> {
+                notificationService.notifyOfConfigError(Bundle.message("error.config.path"))
+            }
+            executablePath.isNullOrBlank() -> {
+                notificationService.notifyOfConfigError(Bundle.message("error.executable.path"))
+            }
+            else -> process = createEditorService(executablePath, configPath)
+        }
+    }
+
+    /**
+     * Shuts down the editor service and destroys the process.
+     */
+    fun destroy() {
+        val stdout = this.process?.inputStream
+        val stdin = this.process?.outputStream
+        val stderr = this.process?.errorStream
+        LOGGER.info(Bundle.message("editor.service.shutting.down"))
+        stdin?.close()
+        LOGGER.info(stdout?.bufferedReader().use { it?.readText() })
+        stdout?.close()
+        LOGGER.info(stderr?.bufferedReader().use { it?.readText() })
+        stderr?.close()
+        this.process?.destroy()
+        this.process = null
+    }
+
+    private fun createEditorService(executablePath: String, configPath: String): Process {
+        val pid = ProcessHandle.current().pid()
+        val commandLine = GeneralCommandLine(
+            executablePath,
+            "editor-service",
+            "--config",
+            configPath,
+            "--parent-pid",
+            pid.toString()
+        )
+
+        val workingDir = File(configPath).parent
+
+        when {
+            workingDir != null -> {
+                commandLine.withWorkDirectory(workingDir)
+                LOGGER.info(Bundle.message("editor.service.starting", executablePath, configPath, workingDir))
+            }
+            else -> LOGGER.info(Bundle.message("editor.service.starting.working.dir", executablePath, configPath))
+        }
+
+        return commandLine.createProcess()
+    }
+
+    private fun getProcess(): Process {
+        return process
+            ?: throw ProcessUnavailableException(Bundle.message("editor.process.cannot.get.editor.service.process"))
+    }
```

Comment:
This is the only new thing in this file. Everything else is copy paste from the old `DprintService`